cms/server/admin/handlers/dataset.py

```
@@ -1310,3 +1318,344 @@ def post(self, dataset_id, generator_id):
+                "Unknown operation: %s" % operation,
+            )
+            self.redirect(fallback_page)
+            return
+
+        # Gather testcases
+        testcases = []
+        for id_str in id_strings:
+            try:
+                tid = int(id_str)
+            except ValueError:
+                raise tornado.web.HTTPError(400)
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)